### PR TITLE
feat(cli): add agent commands

### LINKED ISSUE #3901: Medplum Agent Commands exposed in CLI
## Use Case
Some DevOps engineers primarily work with CLI tooling and shell scripts, rather than rest API calls. 

To help them, we want to expose all the Agent maintenance commands as CLI commands. This will also help to integrate into monitoring tools such as [BetterStack](https://betterstack.com/).

## Proposal

```
medplum agent push <args..> [--wait-for-response]
medplum agent ping <host> [pingCount]
medplum agent status <agentId>
medplum agent bulk-status [criteria]

# When added:
medplum agent reload-config <agentId1>[agentId2..agentIdN]
medplum agent upgrade <version> <agentId1>[agentId2..agentIdN]
```

### PR DESCRIPTION
**Pull request by ThatOneBro**
Closes #3901 and #4681

### CODE REVIEW

**Comment by reshmakh** [comment]
Planned for today

File: packages/cli/src/agent.ts [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+    let pingResult: string;
+    try {
+      pingResult = (await medplum.pushToAgent(agentRef, ipOrDomain, `PING ${count}`, ContentType.PING, true, {
+        maxRetries: 0,
+      })) as string;
+    } catch (err) {
+      throw new Error('Unexpected response from agent while pinging', { cause: err });
+    }
+
+    console.info(pingResult);
```
Comment (mattwiller) [inline packages/cli/src/agent.ts:124]: Suggest tucking statements inside `try` for cohesion:
```suggestion
    try {
      const pingResult = (await medplum.pushToAgent(agentRef, ipOrDomain, `PING ${count}`, ContentType.PING, true, {
        maxRetries: 0,
      })) as string;
      console.info(pingResult);
    } catch (err) {
      throw new Error('Unexpected response from agent while pinging', { cause: err });
    }
```

File: packages/cli/src/agent.ts [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+      severity: issue.severity,
+      code: issue.code,
+      details: issue.details?.text ?? 'No details to show',
+    } satisfies FailedRow;
+    failedRows.push(row);
+  }
+
+  console.info();
+  console.info(`${successfulRows.length} successful response(s):`);
+  console.info();
```
Comment (mattwiller) [inline packages/cli/src/agent.ts:304]:
```suggestion
  console.info(`\n${successfulRows.length} successful response(s):\n`);
```

File: packages/cli/src/agent.test.ts [hunk 3]
```diff
@@ -0,0 +1,1326 @@
...
+    jest.clearAllMocks();
+    medplum = new MockClient();
+    (createMedplumClient as unknown as jest.Mock).mockImplementation(async () => medplum);
+
+    process.env = { ...env };
+    medplumGetSpy = jest.spyOn(medplum, 'get');
+  });
+
+  afterEach(() => {
+    process.env = env;
```
Comment (mattwiller) [inline packages/cli/src/agent.test.ts:69]: Are both these assignments to `process.env` needed?

File: packages/cli/src/agent.test.ts [hunk 4]
```diff
@@ -0,0 +1,1326 @@
...
+                { name: 'status', valueString: 'unknown' },
+                { name: 'version', valueString: 'unknown' },
+              ],
+            } satisfies Parameters,
+          ];
+        });
+
+        await expect(main(['node', 'index.js', 'agent', 'status', agentId])).resolves.toBeUndefined();
+        expect(medplumGetSpy).toHaveBeenCalledWith(
+          // NOTE(ThatOneBro 28 Sept 2024):
```
Comment (mattwiller) [inline packages/cli/src/agent.test.ts:99]: 👍

**Review by mattwiller — APPROVED** [review]

File: packages/cli/src/agent.test.ts (hunk 3 above)
Comment (ThatOneBro) [reply packages/cli/src/agent.test.ts:69]: I think this `afterEach` should actually be an `afterAll` probably, to reset the `process.env` at the end

**Pull request merged**